rca/shortcourses/tests/test_access_planit.py

```diff
@@ -0,0 +1,217 @@
+                    "box/clientinput/course/coursebooker.aspx?coursedateid=12085"
+                ),
+                "enquire_url": (
+                    "https://rca.accessplanit.com/accessplansand"
+                    "box/clientinput/company/contactcompany.aspx?contacttype=8"
+                    "&coursecalid=12085"
+                ),
+            },
+            {
+                "course_date_id": "12071",
+                "course_id": "731014",
+                "cost": "1250",
+                "start_date": datetime.datetime(2020, 7, 20, 9, 0),
+                "end_date": datetime.datetime(2020, 7, 24, 17, 0),
+                "spaces_available": "20",
+                "status": "Available",
+                "book_now_url": (
+                    "https://rca.accessplanit.com/accessplansand"
+                    "box/clientinput/course/coursebooker.aspx?coursedateid=12071"
+                ),
+                "enquire_url": (
+                    "https://rca.accessplanit.com/accessplansand"
+                    "box/clientinput/company/contactcompany.aspx?contacttype=8"
+                    "&coursecalid=12071"
+                ),
+            },
+        ]
+        self.query = QueryDict(mutable=True)
+        self.query.update(
+            {
+                "CompanyID": "ROYALC9RCH",
+                "courseIDs": 1,
+                "venueIDs": "",
+                "categoryIDs": "",
+            }
+        )
+        self.query = self.query.urlencode()
+
+    def test_xml_fetch(self):
+        """ Test the XML fetch responds. """
+        response = requests.get(
+            settings.ACCESS_PLANIT_XML_BASE_URL + self.query, timeout=5
+        )
+        self.assertEqual(response.status_code, 200)
+
+    def test_xml_fetch_no_venue(self):
+        """ Prove that you must pass blank values as parameters"""
+        query = QueryDict(mutable=True)
+        query.update({"CompanyID": "ROYALC9RCH", "courseIDs": 1})
+        query = query.urlencode()
+        response = requests.get(settings.ACCESS_PLANIT_XML_BASE_URL + query, timeout=5)
+        self.assertEqual(response.text, """Missing parameter: venueIDs.\r\n""")
+        self.assertEqual(response.status_code, 500)
+
+    """ Patch the request module totally to force the timeout so we can test the
+        result of the try/expect.
+
+        Test here that data returned from a timeout is a empty dict
+    """
+
+    @mock.patch("rca.shortcourses.access_planit.requests.get")
+    def test_data_if_timeout(self, mock_get):
+        """ If a timeout is caught the xml_data should be an empty list"""
+        # logging.disable(logging.CRITICAL)  # Don't log exceptions for tests... is this bad?
+        mock_get.side_effect = Timeout
+        data = AccessPlanitXML(course_id=1)
+        xml_data = data.get_data()
+        self.assertEqual(xml_data, [])
+
+    def test_management_command_fetch_data(self):
+        """ Test that xml with no dates goes in the cache as [] """
```

Comment:
Noting that we discussed setting the cache to `[]` with no dates can be tested outside of the management command, and that testing the command could consist of mocking `AccessPlanitXML` and checking how its used.

It might still be worth having an integration test that deliberately isn't isolated in this way in order to test running the command and the eventual outcome.

Reply:
Thanks Rich, I've added a test that just instatiates the AccessPlanitXML class with a course id that would cause this and tested the return for the data = [].